### PR TITLE
tickets/DM-19817 Create framework for runing ci_hsc tests with gen3

### PR DESCRIPTION
**Pull request by natelust**

### CODE REVIEW

File: bin.src/ingestExternalData.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    for entry in products:
+        dsType = DatasetType(entry.datatype, entry.dimensions, entry.storageClass)
+        butler.registry.registerDatasetType(dsType)
+        dsRef = butler.registry.addDataset(dsType, entry.dataId, butler.run)
+        butler.datastore.ingest(entry.path, dsRef)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Writes Curtaled Calibrations")
```
Comment (hsinfang) [inline bin.src/ingestExternalData.py]: Update this description?

File: bin.src/ingestRaws.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import lsst.log
+from lsst.log import Log
+
+from lsst.daf.butler import Butler
+from lsst.obs.base.gen3 import RawIngestTask, RawIngestConfig
+
+import lsst.obs.subaru.gen3.hsc.instrument
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Writes Curtaled Calibrations")
```
Comment (hsinfang) [inline bin.src/ingestRaws.py]: This too

File: bin/pipeline.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+lsst.pipe.tasks -i calib,shared/ci_hsc -o "$COLLECTION" run \
+--register-dataset-types \
+-t isrTask.IsrTask:isr -C isr:"$CI_HSC_GEN3_DIR"/configs/isr.py \
+-t characterizeImage.CharacterizeImageTask:cit -C cit:"$CI_HSC_GEN3_DIR"/configs/charImage.py \
+-t calibrate.CalibrateTask:ct -C ct:"$CI_HSC_GEN3_DIR"/configs/calibrate.py \
+-t makeCoaddTempExp.MakeWarpTask:mwt -C mwt:"$CI_HSC_GEN3_DIR"/configs/makeWarp.py \
+-t assembleCoadd.CompareWarpAssembleCoaddTask:cwact -C cwact:"$CI_HSC_GEN3_DIR"/configs/compareWarpAssembleCoadd.py \
+-t multiBand.DetectCoaddSourcesTask \
+-t mergeDetections.MergeDetectionsTask:mdt -C mdt:"$CI_HSC_GEN3_DIR"/configs/mergeCoaddDetections.py \
+-t deblendCoaddSourcesPipeline.DeblendCoaddSourcesSingleTask \
```
Comment (hsinfang) [inline bin/pipeline.sh]: Would it make sense to add config for DetectCoaddSourcesTask and DeblendCoaddSourcesSingleTask too, just to be more future proof?

File: configs/forcedPhotCcd.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import os
+
+from lsst.utils import getPackageDir
+
+subaruConfig = os.path.join(getPackageDir("obs_subaru"), "config", "forcedPhotCcd.py")
+if os.path.exists(subaruConfig):
+    config.load(subaruConfig)
+hscConfig = os.path.join(getPackageDir("obs_subaru"), "config", "hsc", "forcedPhotCcd.py")
+if os.path.exists(hscConfig):
+    config.load(hscConfig)
```
Comment (hsinfang) [inline configs/forcedPhotCcd.py]: In the current `ci_hsc`  the config override from https://github.com/lsst/ci_hsc/blob/master/forcedPhotCcdConfig.py
has `config.references.skipMissing = True`. Do we need that added here?

File: configs/safeClipAssembleCoadd.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import os
```
Comment (hsinfang) [inline configs/safeClipAssembleCoadd.py]: This file seems unused and not needed

File: configs/assembleCoadd.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import os
```
Comment (hsinfang) [inline configs/assembleCoadd.py]: This file seems unused and not needed

File: ups/ci_hsc_gen3.table [hunk 7]
```diff
@@ -0,0 +1 @@
+setupRequires(ci_hsc)
```
Comment (hsinfang) [inline ups/ci_hsc_gen3.table]: Perhaps `testdata_ci_hsc` instead of `ci_hsc`?

File: resources/calibs.sql [hunk 8]
```diff
@@ -0,0 +1,2365 @@
+PRAGMA foreign_keys=OFF;
```
Comment (hsinfang) [inline resources/calibs.sql]: Is this a stopgap before we have some calibration-ingestion utilities?

File: bin/linker.sh [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
```
[the remark this answers is not in the document]
Comment (natelust) [reply bin/linker.sh]: The scons --clean option does this for you

File: configs/forcedPhotCcd.py (hunk 4 above)
Comment (natelust) [reply configs/forcedPhotCcd.py]: I think that does not apply to us, as we are using the preflight to decide on what to run on.

File: resources/calibs.sql (hunk 8 above)
Comment (natelust) [reply resources/calibs.sql]: Yes, exactly, we can toss it away with the associated script when we are ready.

**Pull request merged**